diag_manager/diag_manager.F90

```diff
@@ -584,14 +590,26 @@ INTEGER FUNCTION register_diag_field_array(module_name, field_name, axes, init_t
     END IF
 
     if (use_modern_diag) then
-            call diag_object_placeholder(1)%register &
-       (module_name, field_name, axes, init_time, &
-       long_name, units, missing_value, Range, mask_variant, standard_name, &
-       do_not_log, err_msg, interp_method, tile_count, area, volume, realm) !(no metadata here)
+      !! Create a diag object, initialize it with the registered data, and insert
+      !! it ino the diag_obj_container singleton.
+
+      allocate( diag_obj )
+      call diag_obj%register (module_name, field_name, axes, init_time, &
+        long_name, units, missing_value, Range, mask_variant, standard_name, &
+        do_not_log, err_msg, interp_method, tile_count, area, volume, realm) !(no metadata here)
+
+      diag_obj_ptr => diag_obj
+      status_ic = the_diag_object_container%insert(diag_obj_ptr%get_id(), diag_obj_ptr)
+      if(status_ic .ne. 0) then
+         print *, "Insertion ERROR for id ", diag_obj_ptr%get_id()
+      endif
     endif
 
+    !! ***  ERROR MISSING ENDIF ???  ***  endif
+
   END FUNCTION register_diag_field_array
 
+
   !> @brief Return field index for subsequent call to send_data.
   !! @return field index for subsequent call to send_data.
   INTEGER FUNCTION register_static_field(module_name, field_name, axes, long_name, units,&
```

Comment:
@thomas-robinson @ngs333 

I think static fields are need to be registered and added to the_diag_object_container?

Reply:
@thomas-robinson , @uramirez8707  You mean regestering should be done as part of this PR? I would prefer a second PR for that and possibly further editions to the diag_object.

Reply:
ah ok.